src/accelerate/utils/modeling.py

```diff
@@ -1919,3 +1899,44 @@ def has_offloaded_params(module: torch.nn.Module) -> bool:
     from ..hooks import AlignDevicesHook  # avoid circular import
 
     return hasattr(module, "_hf_hook") and isinstance(module._hf_hook, AlignDevicesHook) and module._hf_hook.offload
+
+
+@contextlib.contextmanager
+def align_module(module: torch.nn.Module, execution_device: Optional[torch.device] = None):
+    """
+    Moves a module's parameters to the specified execution device.
+
+    Args:
+        module (`torch.nn.Module`):
+            Module with parameters to align.
+        execution_device (`torch.device`, *optional*):
+            If provided, overrides the module's execution device within the context.
+
+    Yields:
+        None: Yields control while the module's parameters are aligned to the execution device.
+    """
+    if has_offloaded_params(module):
+        if execution_device is not None:
+            original_device = module._hf_hook.execution_device
+            module._hf_hook.execution_device = execution_device
+
+        try:
+            module._hf_hook.pre_forward(module)
+            yield
+        finally:
+            module._hf_hook.post_forward(module, None)
+            if execution_device is not None:
+                module._hf_hook.execution_device = original_device
+
+    elif execution_device is not None:
```

Comment:
Just for my understanding: The functionality of this branch was not there before, right? Is this some kind of fix or is it in case the context manager is used in other situations where it may apply?

Reply:
This branch is not a bug fix, `accelerate`'s behavior should be exactly the same in this branch as in main

Reply:
This branch simply introduces a utility function which can help simplify existing code in the `accelerate` package as well as be useful for downstream `accelerate` users, such as [llm-compressor](https://github.com/vllm-project/llm-compressor)

Reply:
Okay, so this is not strictly for use within accelerate in mind, makes sense.